src/core/ddsc/src/dds_serdata_default.c

```diff
@@ -603,8 +603,10 @@ static struct dds_serdata_default *serdata_default_from_sample_cdr_common (const
       }
       break;
     case SDK_DATA:
-      if (!dds_stream_write_sample (&os, sample, &tp->type))
+      if (!dds_stream_write_sample (&os, sample, &tp->type)) {
+        dds_ostream_fini (&os);
```

Comment:
Does this also work when a serdata pointer is used from the freelist (line 570 set `os` to `d`, which can be from the freelist). And does `d` also need to be freed on lines 602 and 612?

Reply:
Thank you @dpotman! Clearly I am no longer as clever as I used to be 😂

It is such a small change that I decided to try again and I simply force-pushed it. I'd appreciate it if you would have another look.